test/extended/util/framework.go

```diff
@@ -1018,32 +1018,60 @@ func CheckBuildCancelled(b *buildv1.Build) bool {
 	return b.Status.Phase == buildv1.BuildPhaseCancelled
 }
 
-// WaitForServiceAccount waits until the named service account gets fully
-// provisioned
+func getServiceAccount(c corev1client.ServiceAccountInterface, name string) (*corev1.ServiceAccount, error) {
+	sc, err := c.Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		if kapierrs.IsNotFound(err) || kapierrs.IsForbidden(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return sc, nil
+}
+
+func serviceAccountContainsPullSecret(sa *corev1.ServiceAccount) bool {
+	hasDockerCfg := false
+	for _, s := range sa.ImagePullSecrets {
+		if strings.Contains(s.Name, "-dockercfg-") {
+			hasDockerCfg = true
```

Comment:
if you choose to refactor like this, this should be `return true`

Reply:
Done